backend/azure/dtmb/internal/utils/typeconversion.go

```diff
@@ -0,0 +1,315 @@
+package utils
+
+import (
+	"fmt"
+
+	"github.com/microsoft/durabletask-go/internal/protos"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+
+	dtmbprotos "github.com/microsoft/durabletask-go/backend/azure/dtmb/internal/backend/v1"
+)
+
+func ConvertFailureDetails(failureDetails *dtmbprotos.FailureDetails) *protos.TaskFailureDetails {
+	if failureDetails == nil {
+		return nil
+	}
+
+	return &protos.TaskFailureDetails{
+		ErrorType:      failureDetails.GetErrorType(),
+		ErrorMessage:   failureDetails.GetErrorMessage(),
+		StackTrace:     &wrapperspb.StringValue{Value: failureDetails.GetStackTrace()}, // what if this is not string
+		InnerFailure:   ConvertFailureDetails(failureDetails.GetInnerFailure()),
+		IsNonRetriable: !failureDetails.GetRetriable(), // need to negate this
+	}
+}
+
+func ConvertTaskFailureDetails(taskFailureDetails *protos.TaskFailureDetails) *dtmbprotos.FailureDetails {
+	if taskFailureDetails == nil {
+		return nil
+	}
+
+	return &dtmbprotos.FailureDetails{
+		ErrorType:    taskFailureDetails.GetErrorType(),
+		ErrorMessage: taskFailureDetails.GetErrorMessage(),
+		StackTrace:   taskFailureDetails.GetStackTrace().GetValue(),
+		InnerFailure: ConvertTaskFailureDetails(taskFailureDetails.GetInnerFailure()),
+		Retriable:    !taskFailureDetails.GetIsNonRetriable(),
+	}
+}
+
+func ConvertEvent(event *dtmbprotos.Event) (*protos.HistoryEvent, error) {
+	switch typedEvent := event.GetEventType().(type) {
+	case *dtmbprotos.Event_ExecutionStarted:
+		return &protos.HistoryEvent{
+			EventId:   int32(event.GetSequenceNumber()), // is this correct?
+			Timestamp: event.GetTimestamp(),
+			EventType: &protos.HistoryEvent_ExecutionStarted{
+				ExecutionStarted: &protos.ExecutionStartedEvent{
+					Name:    typedEvent.ExecutionStarted.GetName(),
+					Version: &wrapperspb.StringValue{Value: typedEvent.ExecutionStarted.GetVersion()},
+					Input:   &wrapperspb.StringValue{Value: string(typedEvent.ExecutionStarted.GetInput())}, // what if this is not string
+					OrchestrationInstance: &protos.OrchestrationInstance{
+						InstanceId:  typedEvent.ExecutionStarted.GetOrchestrationId(),
+						ExecutionId: wrapperspb.String(typedEvent.ExecutionStarted.GetExecutionId()),
+					},
+					ParentInstance: &protos.ParentInstanceInfo{
+						TaskScheduledId: int32(typedEvent.ExecutionStarted.GetParent().GetSequenceNumber()),
+						Name:            &wrapperspb.StringValue{Value: typedEvent.ExecutionStarted.GetParent().GetName()},
```

Comment:
Nit: consider creating a helper method `toStringValue`...

..which accepts `string` and `byte[]` too if you want to make your life easier!

Reply:
@cgillum @ItalyPaleAle should we do a base64 conversion if the type is byte[] ?

Reply:
Given what Chris said about currently us using strings only, I think supporting strings only is fine for now.

Reply:
Yeah sounds good